src/translate.rs

```diff
@@ -97,6 +97,44 @@ pub fn header_items(ctxt: &rustast::ExtCtxt) -> Vec<rustast::P<rustast::Item>> {
 	).unwrap());
 
 	items.push(quote_item!(ctxt,
+		#[derive(PartialEq, Eq, Debug)]
+		pub struct ParseError {
+			pub line: usize,
+			pub column: usize,
+			pub offset: usize,
+			pub expected: ::std::collections::HashSet<&'static str>,
+		}
+	).unwrap());
+
+	items.push(quote_item!(ctxt,
+		impl ::std::fmt::Display for ParseError {
+			fn fmt(&self, fmt: &mut ::std::fmt::Formatter) -> ::std::result::Result<(), ::std::fmt::Error> {
+				fn quote_expect(s: &'static str) -> &'static str {
+					match s {
+						"\n" => "\\n",
+						"\t" => "\\t",
+						_ => s,
+					}
+				}
```

Comment:
Could this use [escape_default](http://doc.rust-lang.org/std/str/trait.StrExt.html#tymethod.escape_default) instead of defining a new escaping function? This one will do the wrong thing for strings that contain escaped characters as a substring, as well as non-ASCII characters.